hamp_radar/iquick.py

```diff
@@ -43,12 +106,57 @@ def main_ofs(mainblock):
 
 
 def get_tag_size(data):
+    """
+    Extracts the tag (also known as signature) and the pointer for size
+    from 'data' assuming layout as in Meteorological Ka-Band Cloud Radar
+    MIRA35 Manual, section 2.3.1 'Definition of chunk common structure'
+    where the tag is the first 4 bytes of data and the pointer is the
+    next 4 bytes.
+
+    Args:
+        data (bytes): The data assumed to conform with Meteorological Ka-Band
+                      Cloud Radar MIRA35 Manual, section 2.3.1 'Definition of
+                      chunk common structure'
+
+    Returns:
+        Tuple[bytes, int]: The tag and the pointer for size.
+    """
     return bytes(data[:4]), data[4:8].view("<i4")[0]
 
 
 def get_geometry(data):
+    """
+    Interprets and returns the geometry of the 'data' assuming it is a
+    memmap of a PDS file (or an open PDS file) for radar IQ data with structure
+    as in Meteorological Ka-Band Cloud Radar MIRA35 Manual, section 2.3.2
+    'File structure'.
+
+    First attempt will read the main tag and size from first 8 bytes
+    of the data. If the data length is insufficient, or the main tag is not
+    found by attempting to return to the start of the file from main_size+8,
+    it shifts the offset by 1024 bytes (to skip file header) and tries again.
+    If the main tag is still not found, it raises a ValueError.
+
+    If get_tag_size is sucessful, the function defines a generator
+    `main_blocks` that iterates through the data, yielding
+    `SingleMainBlockGeometry` instances for each block (i.e. for each main chunk
+    as defined in section 2.3.2)
```

Comment:
This doesn't really describe the external behavior of the function, so I'd remove it from the docstring. The internal behavior can be seen when reading the code.
```suggestion
```